src/lib/routing/page.ts

```diff
@@ -1,36 +1,43 @@
-import type { PageRecord, SiteLocale } from '@lib/types/datocms';
+import type { HomeRouteFragment, PageRouteFragment, SiteLocale } from '@lib/types/datocms';
+
+export type PageRouteForPath =
+  & Pick<PageRouteFragment, '_allSlugLocales' | 'parentPage'>
+  & { parentPage?: PageRouteForPath | null };
+
+export type PageOrHomeRoute =
```

Comment:
@jurgenbelien to me `routing/page.ts` is only about the `Page` model and its routing. If the routing for the `Home` model would've been more complicated, there would've been a `routing/home.ts` as well. Similarly there's a [`routing/file.ts` in #147](https://github.com/voorhoede/head-start/pull/147/files#diff-8fcebe9a1639fde4961c47b9b8e882d8e6a3f39de0744d938e5c6ee86895dfb5). In `routing/index.ts` they all come together. So in that file I would introduce this union type (and also because it's only used there, reducing the mental model). I would consider renaming it `AnyRoute` or `RecordRoute` as with for example the addition of the `File` model it would become `PageOrHomeOrFileRoute` etc.